### PR TITLE
Added some changes

### PR DESCRIPTION
**Pull request by ilanolkies**
- Improved introduction
- Modified state of the art a little bit
- Increased motivation
- Correct indentations
- Set _registration flow_ as required, not optional flow. Set `sdr` as required part of this flow
- Identified repeated on _register_ and _login_. Added section _authenticating requests_ "refactoring" this

### CODE REVIEW

File: ssi/specs/did-auth.markdown [hunk 1]
```diff
@@ -6,6 +6,10 @@ layout: default
 
 A challenge–response authentication is a family of protocols in which one party presents a question ("challenge") and another party must provide a valid answer ("response") to be authenticated. {% include ref.html id="10" %}
 
+DID Auth is a protocol that allows asking the controller of an account to sign a random message, thus demonstrating control of the account at the time of the question. This protocol is used as a login> it allows detecting that the user is in control of the account at the time of access to the application.
+
+Additionally, it allows the application to request specific data at the time of registration, for example the user's email or telephone number. These requested data follow a specific standard, which allows the client to provide it in a unified way and even cryptographically signed by a third party. That is, the application can ask the user for verifiable credentials, identifying them by their type in a unique way.
+
```
Comment (javiesses) [inline ssi/specs/did-auth.markdown:12]:
```suggestion
DID Auth is a protocol that allows asking the controller of an account to sign a random message, thus demonstrating control of the account at the time of the question. This protocol used as a login method checks that the user controls the account at the time of access to the application.

Additionally, it allows the application to request specific data at the time of registration, for example the user's email or phone number. These requested data follow a specific standard, which allows the client to provide it in a unified way and even cryptographically signed by a third party. This means that the application can ask the user for verifiable credentials, identifying them by their type in a unique way.
```

File: ssi/specs/did-auth.markdown [hunk 2]
```diff
@@ -24,100 +28,113 @@ A challenge–response authentication is a family of protocols in which one part
...
 
 ### Motivation
 
-The motivation of this protocol is to provide user centric registration and authentication mechanisms to be used by services of the web 3.0. We identify two main type of services: _open_ (public access) and _permissioned_ (the user must provide certain information to access it). This protocol allows an external service to prove that the user is in control of their did and, optionally, enables the application to ask the user to share specific private information associated to their DID, in order to register them.
-If it is a permissioned service, it may ask for verified user related information (ie: Verifiable Credentials{% include ref.html id="7" %}) and check if the shared information has been issued by reliable entities.
+The motivation of this protocol is to provide user centric registration and authentication mechanisms to be used by services of the web 3.0 considering:
+
+- Users are identified by their unique blockchain addresses
+- Users hold their private information in user-centric services, or even fully-decentralized platforms
+- Services may require specific user's information to provide access to different users
```
Comment (javiesses) [inline ssi/specs/did-auth.markdown:41]:
```suggestion
- Services may require specific user's information to grant access to the service
```

File: ssi/specs/did-auth.markdown [hunk 3]
```diff
@@ -24,100 +28,113 @@ A challenge–response authentication is a family of protocols in which one part
...
+- Services verify the integrity and authenticity of the information shared by the user
+- Services perform different logic depending on the designed authentication model (we call this _the business logic_)
```
Comment (javiesses) [inline ssi/specs/did-auth.markdown:43]:
```suggestion
- Services perform different logic depending on the selected authentication model (we call this _the business logic_)
```

**Review by javiesses — COMMENTED** [review]
Like it, just minor comments

File: ssi/specs/did-auth.markdown (hunk 3 above)
Comment (ilanolkies) [reply ssi/specs/did-auth.markdown:43]: The job of _designing_ an authentication model is much bigger than just _selecting_ one. Usually, when I talk about _models_ I automatically use _design_ word, is something that must be specifically designed for each use case

**Pull request merged**